wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/SyntaxReader.kt

```diff
@@ -195,6 +195,11 @@ class SyntaxReader(
       }
     }
     expect(start < pos) { "expected a word" }
+    if (!allowLeadingDigit) {
+      expect(!data[start].isDigit(), location().copy(column = start - lineStart)) {
+        "field and constant names cannot start with a digit"
```

Comment:
Maybe I could rephrase it so that it doesn't use `field` or `constant` here... This method is used in many places and doesn't really know why `allowLeadingDigit` is set

Reply:
Maybe
1) Throw an exception with a more generic message, indicating that "`allowLeadingDigit` is false but name started with digit",
2) Catch at the call site that expects names to not start with digits,
3) Rethrow with a more specific message?

Reply:
`allowLeadingDigit` isn't public API so I didn't write it in the message.
The trick is to get the proper location as well.

Maybe throwing internal exception and catching it might be nice. I'll think about it.